lib/std/Build/Step/Run.zig

```diff
@@ -628,6 +628,35 @@ fn make(step: *Step, options: Step.MakeOptions) !void {
     var man = b.graph.cache.obtain();
     defer man.deinit();
 
+    if (run.env_map) |env_map| {
+        const KV = struct { []const u8, []const u8 };
+        var kv_pairs = try std.ArrayList(KV).initCapacity(arena, env_map.count());
+        var iter = env_map.iterator();
+        while (iter.next()) |entry| {
+            kv_pairs.appendAssumeCapacity(.{ entry.key_ptr.*, entry.value_ptr.* });
+        }
+
+        std.sort.insertion(KV, kv_pairs.items, {}, struct {
```

Comment:
you picked insertion sort?

Reply:
My thinking is that the number of environment variables to be sorted will be small, so the added overhead of something like pdq isn't worth it as it will just be a slower insertion sort. I can switch it to pdq if we're worried people will make run steps with hundreds of environment variables.

Reply:
"why are we driving into the oncoming traffic lane?"
"well there aren't any cars coming right now, so I thought it would be a good idea. I can steer us back into the normal traffic lane if you're worried people will crash into us."

"worried" is a funny way to categorize my reaction here.

anyway, what do you think the very first line of the pdq sort implementation does?

Reply:
I've switch it to pdq.